src/Cli.ts

```diff
@@ -75,6 +90,41 @@ export class IonCliError {
     }
 }
 
+/** Comparison result types for the comparison report */
+export enum ComparisonResultType {
+    EQUAL = "EQUAL",
+    NOT_EQUAL = "NOT_EQUAL",
+    ERROR = "ERROR"
+}
+
+export class IonComparisonReport {
+    result: ComparisonResultType;
+    lhs: ComparisonContext;
+    rhs: ComparisonContext;
+    message: string;
+
+    constructor(result: ComparisonResultType, lhs: ComparisonContext, rhs: ComparisonContext, message: string) {
+        this.result = result;
+        this.lhs = lhs;
+        this.rhs = rhs;
+        this.message = message;
+    }
+
+    writeErrorReport(comparisonReportFile: WriteStream| NodeJS.WriteStream, event_index) {
+        let writer = makeTextWriter();
+        writer.stepIn(IonTypes.STRUCT);
+        writer.writeFieldName('result');
+        writer.writeSymbol(this.result);
+        this.lhs.writeComparisonContext(writer, true, event_index);
+        this.rhs.writeComparisonContext(writer, false, event_index);
```

Comment:
I recommend just setting the field names here instead of in `writeComparisonContext`.

Reply:
Should I move `writeComparisonContext` method here in `IonComparisonReport`? So that for `lhs` and `rhs` I won't have to write the same lines of code twice.

Reply:
No, I think `writeComparisonContext` is fine as a method of `ComparisonContext`. Which lines of code are you referring to that you'd have to repeat?

Reply:
I think I misunderstood what you mean by setting field names here. I got it now. will change that. Thanks.